op-service/txmgr/txmgr.go

```diff
@@ -126,8 +137,52 @@ type TxCandidate struct {
 // The transaction manager handles all signing. If and only if the gas limit is 0, the
 // transaction manager will do a gas estimation.
 //
-// NOTE: Send should be called by AT MOST one caller at a time.
+// NOTE: Send can be called concurrently, the nonce will be managed internally.
 func (m *SimpleTxManager) Send(ctx context.Context, candidate TxCandidate) (*types.Receipt, error) {
+	m.resetWG.Wait() // wait for any current resets
+
+	receipt, err := m.send(ctx, candidate)
+	if err != nil {
+		m.reset()
+	}
+	return receipt, err
+}
+
+// reset the transaction manager. All currently pending transactions will receive
+// a ResetErr error. This is called if any pending send returns an error.
+func (m *SimpleTxManager) reset() {
```

Comment:
I just had another idea here that requires a little more thinking about the API. We could use an `ErrGroup` to manage the resets (because if you use `group.WithContext` any failure will propagate to the other transactions which should cancel them. Then you'll need to figure out how to reset the nonce (not sure if it should be done internally or externally).

There's another open question about if the `ErrGroup` is inside the `SimpleTxManager` or inside `TxQueue` (errgroups also have a semaphore like functionality). If the errgroup is inside the `TxQueue` there's a last question of how to handle the reset (internal vs external vs the queue taking the simpleTxMgr or taking an interface)